lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -235,11 +235,11 @@ <h3 class="course-title" id="course-title-${enrollment.course_id}">
                 % endif
             % endif
 
-            ## Right now, the gear dropdown for entitlements only contains the 'unenroll' link, so we should hide the
-            ## gear altogether if the user is unable to unenroll/refund their entitlement. Later, when we add more options
-            ## to the gear dropdown, we can remove this check.
-            % if entitlement and can_refund_entitlement:
-                <%include file='_dashboard_entitlement_actions.html' args='course_overview=course_overview,entitlement=entitlement,dashboard_index=dashboard_index, can_refund_entitlement=can_refund_entitlement'/>
+            ## We should only show the gear dropdown if the user is able to refund/unenroll from their entitlement
+            ## and/or if they have selected a course run and email_settings are enabled
+            ## as these are the only actions currently available
+            % if entitlement and (can_refund_entitlement or (entitlement_session and show_email_settings)):
```

Comment:
How does show_email_settings get set?

Reply:
Do we have any logic anywhere to prevent the emails settings from being shown when the entitlement doesn't have an associated enrollment?

Reply:
If the session_id for the entitlement is in the `show_email_settings_for` list, which is set [here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/views.py#L840-L845).

and we only check this variable for entitlements when there is an enrollment set for the entitlement, so we know it won't be using data from an arbitrarily chosen course run.

Reply:
`entitlement_session` is `None` if the entitlement doesn't have an associated enrollment, so the if statement on line 46 of `lms/templates/dashboard/_dashboard_entitlement_actions.html` would be false and not include the email settings

Reply:
Could you simply this by including the `entitlement_session` check in the logic that computes `show_email_settings`?